hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/sender/googlebigquery/avro/conversions/ConvertAndCastPrimitivesTest.groovy

```diff
@@ -0,0 +1,94 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro.conversions
+
+import com.google.protobuf.DynamicMessage
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.generic.GenericRecordBuilder
+import org.junit.jupiter.api.Test
+import pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro.AvroTrait
+import pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro.GoogleBigQueryAvroToProtoConverter
+import pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro.descriptor.BoolProto
+import pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro.descriptor.DoubleProto
+import pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro.descriptor.FloatProto
+import pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro.descriptor.Int32Proto
+import pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro.descriptor.Int64Proto
+import pl.allegro.tech.hermes.consumers.consumer.sender.googlebigquery.avro.descriptor.StringProto
+import spock.lang.Specification
+
+class ConvertAndCastPrimitivesTest  extends Specification  implements AvroTrait{
+
+    @Test
+
+    void 'should convert and cast primitive types'() {
+
+        given:
+        Schema schema = getSchemaFromResources("${suite}/${avroType}")
+        String fieldName = "field"
+        GenericRecord record = new GenericRecordBuilder(schema).set(fieldName, avroValue).build()
+        GoogleBigQueryAvroToProtoConverter converter = new GoogleBigQueryAvroToProtoConverter()
+
+        when:
+        DynamicMessage message = converter.convertToProtoMessage(protoDescriptor, record)
+
+
+        then:
+        message != null
+        println(message)
+        def field = message.getDescriptorForType().findFieldByName(fieldName)
+        def converted_value = message.getField(field)
+        if (field.getDefaultValue() != avroValue) {
+            record.hasField(fieldName) == message.hasField(field)
+        }
+        transformResult(converted_value) == expectedProtoValue
+
+
+        where:
+        suite        | avroType  | protoDescriptor                              | avroValue | expectedProtoValue | transformResult
+        "primitives" | "string"  | StringProto.PrimitivesString.getDescriptor() | "value" | "value" | { c -> c }
```

Comment:
Why do you need `transformResult` variable if all provided values are the same (identity func)?